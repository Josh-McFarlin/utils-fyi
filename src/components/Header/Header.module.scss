@import "../../../public/static/styles/Colors";

.root {
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  position: sticky;
  top: 0;
  left: 0;
  box-sizing: border-box;
  padding: 8px 16px;
  background: $secondary-bg-color;

  .hamburger {
    margin-right: 16px;
    font-size: 32px;
    color: $primary-text-color;
    cursor: pointer;
  }
}
