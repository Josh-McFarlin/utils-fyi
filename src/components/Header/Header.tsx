import * as React from "react";
import Sidebar from "./Sidebar";
import classes from "./Header.module.scss";

const Header = () => {
  const [showSidebar, setSidebar] = React.useState<boolean>(false);

  const toggleSidebar = () => setSidebar((prev) => !prev);

  return (
    <>
      <Sidebar handleClose={toggleSidebar} open={showSidebar} />
      <div className={classes.root}>
        <div className={classes.hamburger} onClick={toggleSidebar}>
          =
        </div>
        <h1>utils-fyi</h1>
      </div>
    </>
  );
};

export default Header;
