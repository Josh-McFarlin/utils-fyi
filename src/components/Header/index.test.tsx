import React from "react";
import { render } from "../../../test/testUtils";
import Header from "./Header";

describe("Header Component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Header />, {});

    expect(asFragment()).toMatchSnapshot();
  });
});
