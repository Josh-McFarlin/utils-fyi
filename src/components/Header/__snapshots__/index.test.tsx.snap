// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="root hidden"
    role="button"
    tabindex="0"
  >
    <div
      class="sidebar"
      role="button"
      tabindex="0"
    >
      <h1
        class="logo"
      >
        utils-fyi
      </h1>
      <button
        class="route"
        tabindex="0"
      >
        Home
      </button>
      <button
        class="route"
        tabindex="0"
      >
        UUID Generator
      </button>
    </div>
  </div>
  <div
    class="root"
  >
    <button
      class="hamburger"
    >
      =
    </button>
    <h1>
      utils-fyi
    </h1>
  </div>
</DocumentFragment>
`;
